powermon/libs/apicoordinator.py

```diff
@@ -12,32 +16,31 @@ class ApiCoordinator:
     def __str__(self):
         if not self.enabled:
             return "ApiCoordinator DISABLED"
-        return f"ApiCoordinator: adhocTopic: {self.adhocTopic}, announceTopic: {self.announceTopic}"
+        return f"ApiCoordinator: adhocTopic: {self.adhoc_topic_format}, announceTopic: {self.announce_topic}"
 
     @classmethod
-    def fromConfig(cls, config=None, device=None, mqtt_broker=None):
+    def from_config(cls, config=None, device=None, mqtt_broker=None):
         log.debug(f"ApiCoordinator config: {config}")
         if not config:
             log.info("No api definition in config")
-            adhocTopic = "powermon/adhoc"
-            announceTopic = "powermon/announce"
-            refreshInterval = 60
+            refresh_interval = 60
             enabled = False
         else:
-            adhocTopic = config.get("adhoc_topic", "powermon/adhoc")
-            announceTopic = config.get("announce_topic", "powermon/announce")
-            refreshInterval = config.get("refresh_interval", 60)
+            refresh_interval = config.get("refresh_interval", 60)
             enabled = config.get("enabled", True)  # default to enabled if not specified
 
-        return cls(adhocTopic=adhocTopic, announceTopic=announceTopic, enabled=enabled, refreshInterval=refreshInterval, device=device, mqtt_broker=mqtt_broker)
+        adhoc_topic_format = "powermon/{device_id}/addcommand"
+        announce_topic = "powermon/announce"
```

Comment:
these should probably be configurable

Reply:
I thought about that, it needs to be the same in the API so I put some of the config in the announce payload. It's not really clear to me where it should go though. Unless you have an idea where it should go I'll leave it hardcoded for a bit.

I could move it to the DTOs which are shared between the two.

Reply:
what about adding a section to the config file, was playing with it in a fork of your work (to see how it works)
eg:
```yaml
api:
  host: "0.0.0.0"
  port: 5000
  log_level: "info"
  announce_topic: powermon/announce
  adhoc_topic: powermon/adhoc
  refresh_interval: 60
```